sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c

```diff
@@ -237,39 +237,69 @@ static void _send(gnrc_pktsnip_t *pkt, const gnrc_pktsnip_t *orig_pkt)
     }
 }
 
+static gnrc_pktsnip_t *_check_ipv6_hdr(const gnrc_pktsnip_t *orig_pkt)
+{
+    /* discarding const qualifier is safe here */
+    gnrc_pktsnip_t *ipv6 = gnrc_pktsnip_search_type((gnrc_pktsnip_t *)orig_pkt,
+                                                    GNRC_NETTYPE_IPV6);
+    const ipv6_hdr_t *ipv6_hdr = ipv6->data;
+    assert(ipv6 != NULL);
+
+    if (ipv6_addr_is_unspecified(&ipv6_hdr->src)) {
+        ipv6 = NULL;
+    }
+    return NULL;
```

Comment:
Shouldn't this be `ipv6` instead of `NULL`?

Reply:
Can only quote myself here ...

> Note to self, compile test code from yesterday night, even if you think you did...

Reply:
Fixed